src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/user';
+import { Todo } from './types/todo';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number): User | null => (
+  usersFromServer.find((user) => (user.id === userId)) || null
+);
+
+const getUserByName = (userName: string): User | null => (
+  usersFromServer.find((user) => (user.name === userName)) || null
+);
+
+const preparedTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserValid, setIsUserValid] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value.replace(/[^a-zA-Zа-яА-Я0-9\s]/g, ''));
+    setIsTitleValid(Boolean(value));
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setSelectedUser(value);
+    setIsUserValid(Boolean(value));
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setIsTitleValid(Boolean(title.trim()));
+    setIsUserValid(Boolean(selectedUser));
+
+    if (!title.trim() || !selectedUser) {
+      return;
+    }
+
+    setTodos((currentTodos) => {
+      const maxTodoId = Math.max(...currentTodos.map(todo => todo.id));
+      const newUser = getUserByName(selectedUser);
```

Comment:
Use ID as value in select and find user with your helper `getUserById`
That helps you to remove function `getUserByName`.